content/learn/book/plugin-development/_index.md

```diff
@@ -0,0 +1,140 @@
+
+## Licensing
+
+Bevy is dual licensed under [MIT or Apache 2.0](https://www.rust-lang.org/policies/licenses), at your option. Most other Rust projects (including Rust itself) also use this dual-license approach. MIT-only is very popular, and you might be tempted to just use that (Bevy also used to be MIT-only), but there are [very good reasons](https://github.com/bevyengine/bevy/issues/2373) to include both licenses. We highly recommend using the dual MIT / Apache 2.0 license for your Bevy Plugins and crates:
+
+- Including the Apache 2.0 license option significantly reduces the difficulty and boilerplate of proper license compliance in published games because you only need to include one copy of the Apache 2.0 license.
+- Provides maximum compatibility with Bevy and Rust, making it easier to upstream your changes.
+
+## Rust API and Cargo SemVer Guidelines
+
+While they are only guidelines, it can be useful for you to look at and consider the [Rust API guidelines](https://rust-lang.github.io/api-guidelines/) and [Cargo SemVer compatibility conventions](https://doc.rust-lang.org/cargo/reference/semver.html) for recommendations on how to write your API and what to consider a breaking or compatible change.
+
+## Generic Plugin Types
+
+It can be useful to allow your users to supply generic types to your plugins. It can enable them to write custom logic for components to be used; give your plugin a marker component to note an entity it should do some logic to; add events that your plugin should listen for; or a resource your plugin should use (which is useful if you want to apply your plugin to multiple resources of the same type via type aliases.)
+
+You can define a generic plugin like so:
+
+```rust
+// example with a generic type that implements Component
+
+pub struct YourPlugin<T: Component> {
+  pub phantom_t: PhantomData<T>,
+}
+
+impl<T: Component> Plugin for YourPlugin<T> {
+  fn build(&self, app: &mut App) {
+    app.add_systems(Startup, example_function::<T>);
+  }
+  
+  // ... your other logic ...
+}
+
+// example function using your generics
+pub fn example_function<T: Component>(mut commands: Commands) {
+  commands.spawn(T);
+  // ... any other logic here ...
+}
+```
+
+A prime example of generic plugins in use is the [Bevy Cellular Automaton Plugin](https://github.com/ManevilleF/bevy_life).
+
+## Small Crate Size
+
+To avoid long build times in your plugin and in projects using it, you should aim for a small crate size:
+
+- Disable Bevy features that you don't use:
+  - Features are additive => Bevy features enabled in your plugin cannot be disabled by someone using your plugin
+  - You can find Bevy's features [here](https://github.com/bevyengine/bevy/blob/main/docs/cargo_features.md).
```

Comment:
I think this should explicitly mention the usage of`default-features = false` on the bevy dependency, as this could easily be forgotten.